File: FindPeakElement.swift
//After class
// Time Complexity : (O)logn
// Space Complexity : (O)1
// Did this code successfully run on Leetcode : No test case failed for input [1,2]. i will check this later.
import Foundation
class BinarySearch {
static func findPeakElement(_ nums: [Int]) -> Int {
       //edge
       if nums.count == 0 {
           return -1
       }
       
       var low = 0
       var high = nums.count - 1
       while low <= high {
           let mid = low + (high - low)/2
           if mid == 0 || nums[mid] > nums[mid-1] && mid == nums.count - 1 || nums[mid] > nums[mid+1] {
                  return mid
           } else if mid > 0 && nums[mid - 1] > nums[mid] {
               high = mid - 1
           } else {
               low = mid + 1
           }
       }
       return -1
   }
}

File: FindPeakElement.js
// Time Complexity : O(logn)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes

/**
 * @param {number[]} nums
 * @return {number}
 */
var findPeakElement = function(nums) {
  if (nums.length === 0 || nums === null) {
    return -1;
  }

  var low = 0;
  var high = nums.length - 1;

  while (low <= high) {
    var mid = Math.floor((high + low) / 2);
    if (
      (mid === low || nums[mid] > nums[mid - 1]) &&
      (mid === high || nums[mid] > nums[mid + 1])
    ) {
      return mid;
    } else {
      if (nums[mid + 1] > nums[mid]) {
        //move right
        low = mid + 1;
      } else {
        high = mid - 1;
      }
    }
  }
  return -1;
};
